datasets/id/dttot/crawler.py

```diff
@@ -0,0 +1,103 @@
+from zavod import helpers as h
+from zavod import Context
+from lxml.etree import Element
+from datetime import datetime
+from openpyxl import load_workbook
+import xlrd
+import re
+
+DATETIME_FORMAT = "%Y%m%d%H%M%S"
+
+
+def is_sanctions_xlsx(a: Element) -> bool:
+    if (
+        a.text is not None
+        and a.text.strip().lower().find("dttot") > -1
+        and (parse_link(a).endswith(".xls") or parse_link(a).endswith(".xlsx"))
+    ):
+        return True
+    return False
+
+
+def parse_link(a: Element) -> str:
+    return a.get("href", "").split("/")[-1]
+
+
+def find_last_link(context: Context, html: Element) -> Element:
+    last_time = datetime.strptime("19700101000000", DATETIME_FORMAT)
+    last_link = None
+    for link in html.iterdescendants("a"):
+        if not is_sanctions_xlsx(link):
+            continue
+        try:
+            ts = parse_link(link).split(".")[0]
+            ts = datetime.strptime(ts, DATETIME_FORMAT)
+        except Exception as e:
+            context.log.error(f"Could not parse timestamp from {link.get('href')}")
+            continue
+        if ts > last_time:
+            last_time = ts
+            last_link = link
+    return last_link
+
+
+def value_or_none(row, headers, key):
+    try:
+        return row[headers[key]].value
+    except Exception:
+        return None
+
```

Comment:
If an expected key is missing, it would be nice if this warns rather than silently carries on, so that we discover if they change the format and we potentially end up losing an important value.

Reply:
We'll be using a dict instead, fixing this.